Parameter metadata: add the capability to use parameter description (#89)
north-road/qgis-processing-r: website/pages/script-syntax.md

`##X=Field Layer` specifies that variable `X` will be field name taken from `Layer`.

##### Enum

The basic enum syntax is `##var_enum=enum a;b;c` to select from values `a`, `b` or `c`. The value of `var_enum` in this case will be integer indicated position of the selected item in a list. So for example, if `a` is selected the value of `var_enum` will be `0`.

The approach described above works well for a wide range of applications but for **R** it is often not ideal. That is a reason why a new type of enum is available in script syntax.

The syntax is `##var_enum_string=enum literal a;b;c`. The important part here is the keyword `literal` (or more precisely `enum literal`) which specifies that the value from the select box to `var_enum_string` should be passed as a string. So if `b` is selected, then the value of `var_enum_string` will be `"a"`.

#### QGIS definitions used in description files

The inputs to R script are specified as: `QgsProcessingParameter|name|description|defaultValue|other_parameters_separated_by_pipe`.

> The _other_ parameter_ separated_ by_ pipe_ can contain the `QgsProcessingParameter` specific parameters.

Reviewer: Insert some ` around ?

> This metdata line is based on the QGIS definitions used in description file used by GRASS7, SAGA and others providers.

Reviewer: ```suggestion
This metadata line is based on the QGIS definitions used in description file used by GRASS7, SAGA and others providers.
```

So the inputs can look like this:

> `##QgsProcessingParameterFeatureSource|INPUT|Points|0|None|False` specifies that there will be varaible `INPUT` that will be a vector.

Reviewer: ```suggestion
`##QgsProcessingParameterFeatureSource|INPUT|Points|0|None|False` specifies that there will be variable `INPUT` that will be a vector.
```

> `##QgsProcessingParameterField|FIELDS|Attributes|None|INPUT|-1|False|False` specifies taht there will be varaible `FIELDS` that will be a field index.

Reviewer: ```suggestion
`##QgsProcessingParameterField|FIELDS|Attributes|None|INPUT|-1|False|False` specifies that there will be variable `FIELDS` that will be a field index.
```

> `##QgsProcessingParameterRasterLayer|INPUT|Grid|None|False` specifies that there will be varaible `INPUT` that will be a raster.

Reviewer: ```suggestion
`##QgsProcessingParameterRasterLayer|INPUT|Grid|None|False` specifies that there will be variable `INPUT` that will be a raster.
```
